kafka/metrics.go

```diff
@@ -111,20 +102,24 @@ func (h *metricHooks) OnProduceRecordUnbuffered(r *kgo.Record, err error) {
 	}
 
 	if err != nil {
-		errorType := attribute.String("error", "other")
+		errorTypeAttr := attribute.String("error", "other")
```

Comment:
I just noticed a mismatch between this field name here and the spec we outlined; it should be `error_reason` per the spec. Perhaps instead of "other," it could be `"unknown"`

Reply:
yes, I was planning to address this in a separate PR